### ISSUE TITLE
Linux newlines and comments on Windows

### ISSUE THREAD

**Issue by shviller** [teal-language/tl#607]
If you try to `tl run` a file with Linux line endings while on Windows, the first comment in the file will be treated as EOF. That is, if you have a file with
```lua
print'1' --comment\n
print'2'\n
print'3'\n
```
(`\n` as opposed with `\r\n`)
and you try to run it on Windows with `tl run`, only `1` will be printed.
All other combinations (`\n` on Linux, `\r\n` on Linux, `\r\n` on Windows) work fine.
(Checked using 0.14.1 on Linux and code from `master` on Windows; sorry about the mess)

**Comment by hishamhm** [comment]
Thanks for reporting and testing the combinations!

If I may ask you to do some more diagnostics to help us figure out what's going on...

If you add an error at the bottom of the file (like `print(unknown_variable)`), does Teal still catch it? (I'm wondering if the comment gobbles everything else at compile/check time or at code-generation time)

Also, given that test file with `\n` endings, what output file do you get when you run `tl gen test.tl` on Windows? The generated `test.lua` also contains `\n` endings only, right?

What does running that example output with `\n` lines do when running in plain Lua on Windows?

**Comment by shviller** [comment]
It does indeed seem that the comment eats up the rest of the file (maybe it's related to #606?). Here's what the testing results look like:
1. Teal doesn't catch `print(unknown_variable)` appended at the end of the file.
2. `tl gen` produces a lua file with a single line: `print('1')`. Line ending is `\r\n`.
3. Running this .tl file via vanilla Lua (PUC Lua 5.1 through 5.4, plus LuaJIT 2.1.0-beta3) yields what you'd expect:
```
1
2
3
nil
```

**Comment by hishamhm** [comment]
@shviller are you running `tl` on Windows from source or a binary? If running from source, does changing `io.open(filename, "r")` to `io.open(filename, "rb")` in the `tl.process` fix the behavior?

**Comment by hishamhm** [comment]
According to my tests in the Windows CI, it should be fixed now! Let me know if there are still any problems!

**Comment by shviller** [comment]
Seems to not be fixed on my end. Tested the new version with LuaJIT + lfs_ffi and Lua 5.1.5 from Lua for Windows + lfs 1.4.2, and the only difference from the old one is that `tl gen` now produces a file with `\n` line ending (but, for example, issues 605 and 606 are indeed fixed). Singular, since it's just one line. I added another line to make sure:
```lua
print'1'\n
print'2'\n
--comment\n
print'3'\n
print'4'\n
print(unknown_variable)\n
```
and yup, `tl gen` produces a file that's just a single line:
```lua
print('1'); print('2')\n
```
